src/api/pricer/CoinGecko.ts

```diff
@@ -51,14 +52,16 @@ export default class CoinGecko extends BaseExchangeApi {
       throw Error(`CoinGecko API status ${response.status}`);
     }
 
-    if (!response[sourceCurrency]?.[currencyId]) {
+    if (!response[sourceCurrencyName]?.[targetCurrencyName]) {
       throw Error(
-        `Exchange rate for currency pair ${sourceCurrency}/${currencyId} is not available`
+        `Exchange rate for currency pair ${sourceCurrency}/${targetCurrency} is not available`
       );
     }
 
     // It cannot be undefined due to the previous check
-    const conversionRate = response[sourceCurrency]?.[currencyId] as string;
+    const conversionRate = response[sourceCurrencyName]?.[
+      targetCurrencyName
+    ] as string;
```

Comment:
I know it's not yours, just to make it cleaner, you could swap these and remove the comment.
```suggestion
const conversionRate = response[sourceCurrencyName]?.[
  targetCurrencyName
];

if (!conversionRate) {
  throw Error(
    `Exchange rate for currency pair ${sourceCurrency}/${targetCurrency} is not available`
  );
}
```

or, since we already use the base class (pretty much) only for enforcing error (in _getCurrencyName), we could use something like:
```suggestion
return this._validRate(response[sourceCurrencyName]?.[
  targetCurrencyName
] as string, sourceCurrency, targetCurrency);
```
and in base class
```typescript
protected _validRate(rate: string, sourceCurrency: string, targetCurrency: string): BigNumberJs {
    if (!rate) {
      throw Error(
        `Exchange rate for currency pair ${sourceCurrency}/${targetCurrency} is not available`
      );
    }
    
    return BigNumberJs(rate);
  }
```